File: scripts/state.swift
import SwiftUI

struct CounterView: View {
    // Observable state variables
    @State private var count: Int = 0
    @State private var isEnabled: Bool = true
    @Binding var sharedValue: String
    
    let title = "Counter App"  // Immutable
    
    var body: some View {
        VStack {
            Text(title)
            Text("Count: \(count)")
            
            Button("Increment") {
                count += 1  // State update
                count = count + 1  // Another state update
            }
            .disabled(!isEnabled)
            
            Button("Toggle") {
                isEnabled.toggle()  // State update
            }
            
            TextField("Value", text: $sharedValue)
        }
    }
}

class ViewModel: ObservableObject {
    @Published var errorMessage: String? = nil  // Observable
    @Published var items: [String] = []  // Observable
    
    let appName = "MyApp"  // Immutable
    var isLoading: Bool = false  // Mutable
    
    func updateItems(_ newItems: [String]) {
        items = newItems  // State update
    }
    
    func addItem(_ item: String) {
        items.append(item)  // State update
    }
    
    func clearItems() {
        items.removeAll()  // State update
    }
}

File: scripts/state.kt
package com.example.myapp

import androidx.compose.runtime.*
import androidx.compose.material3.*
import androidx.compose.foundation.layout.*

@Composable
fun CounterScreen() {
    // Observable state variables
    var count by remember { mutableStateOf(0) }
    var isEnabled by remember { mutableStateOf(true) }
    val title = "Counter App"  // Immutable
    
    Column {
        Text(text = title)
        Text(text = "Count: $count")
        
        Button(
            onClick = { 
                count++  // State update
                count = count + 1  // Another state update
            },
            enabled = isEnabled
        ) {
            Text("Increment")
        }
        
        Button(onClick = { isEnabled = !isEnabled }) {  // State update
            Text("Toggle")
        }
    }
}

class ViewModel {
    private val _uiState = MutableStateFlow<UiState>(UiState.Loading)
    val uiState: StateFlow<UiState> = _uiState
    
    var errorMessage: String? = null  // Mutable
    val appName = "MyApp"  // Immutable
    
    fun updateState(newState: UiState) {
        _uiState.value = newState  // State update
    }
}

sealed class UiState {
    object Loading : UiState()
    data class Success(val data: String) : UiState()
}